### ISSUE TITLE
88MW302 not connecting to AWS IoT after proper provisioning of board on local network

### ISSUE THREAD

**Issue by moondt** [marvell-iot/aws_starter_sdk#36]
Using 05 Nov 2015 Getting Started Guide on the Wiki at marvell-iot/aws_starter_sdk.
MacOS, XCode w/ CL Tools, MacPorts, OpenOCD 0.9.0, had minicom on my serial port during the provisioning (tried this with 2 different AWS accounts):
```
#### AWS STARTER DEMO ####
Starting micro-AP aws_starter-27DF
Connected successfully to the configured network
aws shadow connect failed : -8
```
Also confirmed that I can ping the persistent IP address that I've associated with the board on my network.

Regarding the failed connect to AWS, wondering whether the board needed reflash with the latest AWS Starter Kit, I flashed aws_starter_sdk-2015_12_8.blob and tried the provisioning and connection again.  No effect; still received -8 return code from the aws_iot_shadow_connect( ) call.  According to the Amazon IoT Embedded C API, -8 is TCP_CONNECTION_ERROR.

I don't believe this is a firewall issue with my network.  I was able to successfully connect via telnet from my network to my amazonaws URL, both ports 1883 and 8883 SSL.

On the AWS IoT side, I did confirm that a policy exists for this device, and it is associated with the certificate.

I'm just trying to evaluate the board as a viable exploration channel for AWS IoT, not wanting to debug an SDK.  But if you have some tactical guidance on what this issue is and would like to advise steps, I'm available to work them.  Thanks for your kind support.

**Comment by anujdeshpande** [comment]
@moondt You can try updating the mcufw on your board using the latest source code. Simply run
`make APP=sample_apps/aws_starter_demo`.
Once you have done that, you can upload the new .bin that has been created to your board.

` sudo python sdk/tools/OpenOCD/flash.py --mcufw bin/mw302_rd/aws_starter_demo.bin `

Press reset and you should be good to go. 

The blob is not up to date with the source code. This is most likely why you are getting that error. While I fix that you do the above to get things working.

**Comment by moondt** [comment]
Thanks, anujdeshpande.  In anticipation of your answer, I had already installed the arm gcc embedded toolchain, per the instructions here.  I first tried 5.3.1 (2016q1 edition), but realized that the makefiles were expecting the previous version.  I now have 4.9.3 (2015q3 edition) installed, and in my environment PATH.  Compilation is fine, but during linking the build breaking on not finding the libg.a library when making the axf.  Will describe this on a different thread.
